src/styles/getModalStyles.ts

```diff
@@ -1,25 +1,60 @@
-import CONST from '../../CONST';
-import variables from '../variables';
-import themeColors from '../themes/default';
-import styles from '../styles';
+import {CSSProperties} from 'react';
+import {ViewStyle} from 'react-native';
+import {ValueOf} from 'type-fest';
+import {ModalProps} from 'react-native-modal';
+import CONST from '../CONST';
+import variables from './variables';
+import themeColors from './themes/default';
+import styles from './styles';
 
-const getCenteredModalStyles = (windowWidth, isSmallScreenWidth, isFullScreenWhenSmall = false) => ({
-    borderWidth: styles.centeredModalStyles(isSmallScreenWidth, isFullScreenWhenSmall).borderWidth,
-    width: isSmallScreenWidth ? '100%' : windowWidth - styles.centeredModalStyles(isSmallScreenWidth, isFullScreenWhenSmall).marginHorizontal * 2,
-});
+function getCenteredModalStyles(windowWidth: number, isSmallScreenWidth: boolean, isFullScreenWhenSmall = false): ViewStyle {
+    return {
+        borderWidth: styles.centeredModalStyles(isSmallScreenWidth, isFullScreenWhenSmall).borderWidth,
+        width: isSmallScreenWidth ? '100%' : windowWidth - styles.centeredModalStyles(isSmallScreenWidth, isFullScreenWhenSmall).marginHorizontal * 2,
+    };
+}
+
+type ModalType = ValueOf<typeof CONST.MODAL.MODAL_TYPE>;
+
+type WindowDimensions = {
+    windowWidth: number;
+    windowHeight: number;
+    isSmallScreenWidth: boolean;
+    isMediumScreenWidth: boolean;
+    isLargeScreenWidth: boolean;
+};
+
+type GetModalStyles = {
+    modalStyle: ViewStyle;
+    modalContainerStyle: ViewStyle | Pick<CSSProperties, 'boxShadow'>;
```

Comment:
Where does this `boxShadow` comes from? Should we just `ViewStyle | CSSProperties`?

Reply:
Line 288 was erroring with `ViewStyle | CSSProperties`, so I went with `ViewStyle | Pick<CSSProperties, 'boxShadow'>`.